app/test/features/activity/item_widgets/activity_attachment_item_widget_test.dart

```diff
@@ -0,0 +1,158 @@
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/features/activities/widgets/space_activities_section/item_widgets/type_widgets/attachment.dart';
+import 'package:acter_notifify/model/push_styles.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../../../common/mock_data/mock_avatar_info.dart';
+import '../../../helpers/test_util.dart';
+import '../mock_data/mock_activity.dart';
+import '../mock_data/mock_activity_object.dart';
+
+void main() {
+  Future<void> createWidgetUnderTest({
+    required WidgetTester tester,
+    MockActivity? mockActivity,
+  }) async {
+    await tester.pumpProviderWidget(
+      overrides: [
+        memberAvatarInfoProvider.overrideWith(
+          (ref, param) => MockAvatarInfo(
+            uniqueId: param.userId,
+            mockDisplayName: 'User-1',
+          ),
+        ),
+      ],
+      child: Material(
+        child: ActivityAttachmentItemWidget(activity: mockActivity!),
```

Comment:
the entire [`activity_item_widget_test.dart`](https://github.com/acterglobal/a3/pull/2675/files#diff-23a9e415ec7db5d52e7d4ca59a88192c7ff4dcc489829dd17482347134a52e64) can be dropped if these tests would just here use:
```suggestion
        child: ActivityItemWidget(activity: mockActivity!),
```

that feels more appropriate than having a bunch of tests for different switch cases ...

Reply:
Yes true, already separate tests with separate files to test each activity item type with lot more data part.